external_parser/parse_example_binary.cc

```diff
@@ -234,16 +259,21 @@ bool binary_parser::read_regular_msg(io_buf *input,
 
   _example_joiner->on_batch_read();
 
+  return process_next_in_batch(examples);
+}
+
+bool binary_parser::process_next_in_batch(v_array<example *> &examples) {
   while (_example_joiner->processing_batch()) {
```

Comment:
this `while(_example_joiner->processing_batch())` loop was duplicated so refactored into `binary_parser::process_next_in_batch()`